### PR TITLE
Add GItHub Actions workflow

### PR DESCRIPTION
**Pull request by binoue**
This PR switches the CKE CI from CircleCI to GitHub Actions.

Signed-off-by: binoue <banji-inoue@cybozu.co.jp>

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      INSTANCE_NAME: cke-${{ matrix.suite }}-${{ github.run_number }}
+      SUITE: ${{ matrix.suite }}
+      CLUSTER: "cke-cluster.yml"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.go-version }}
+      - name: Set up Cloud SDK for cke
+        uses: google-github-actions/setup-gcloud@master
```
Comment (yokaze) [inline .github/workflows/main.yaml:45]: master branch may be instable. I recommend to use the latest release (v0.2.1)

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        with:
+          project_id: ${{ secrets.NECO_TEST_PROJECT_ID }}
+          service_account_key: ${{ secrets.NECO_TEST_SERVICE_ACCOUNT }}
+          export_default_credentials: true
+      - name: Watch cke logs
+        run: |
+            if [ -f .skip ]; then exit 0; fi
+            ./bin/watch_service cke &
```
Comment (yokaze) [inline .github/workflows/main.yaml:53]: GitHub actions does not allow us to use a background run step, thus the logs are invisible after the step is finished.
We need another run step to display the log.
https://github.com/cybozu-go/moco/blob/main/.github/workflows/ci.yaml#L66

File: .github/workflows/release-images.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+env:
+  go-version: 1.15
+  tag: ${GITHUB_REF#refs/tags/v}
+  prerelease: ${{ contains(github.ref, '-') }}
+jobs:
+  sonobuoy:
+    name: Run sonobuoy
+    runs-on: ubuntu-20.04
+    env:
+      INSTANCE_NAME: cke-sonobuoy-${{ github.run_number }}
```
Comment (yokaze) [inline .github/workflows/release-images.yaml:15]: ditto

File: .github/workflows/release-images.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.go-version }}
+      - name: Set up Cloud SDK for cke
+        uses: google-github-actions/setup-gcloud@master
```
Comment (yokaze) [inline .github/workflows/release-images.yaml:22]: ditto

File: .github/workflows/sonobuoy.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-20.04
+    env:
+      INSTANCE_NAME: cke-sonobuoy-${{ github.run_number }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.go-version }}
+      - name: Set up Cloud SDK for cke
+        uses: google-github-actions/setup-gcloud@master
```
Comment (yokaze) [inline .github/workflows/sonobuoy.yaml:18]: ditto

File: .github/workflows/sonobuoy.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  workflow_dispatch:
+env:
+  go-version: 1.15
+jobs:
+  sonobuoy:
+    name: Run sonobuoy
+    runs-on: ubuntu-20.04
+    env:
+      INSTANCE_NAME: cke-sonobuoy-${{ github.run_number }}
```
Comment (yokaze) [inline .github/workflows/sonobuoy.yaml:11]: ditto

**Review by yokaze — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yaml [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.go-version }}
+      - name: Check static resources
+        run: |
+            cp ./static/resources.go /tmp/resources.go
+            make static
+            cmp /tmp/resources.go ./static/resources.go
+      - run: docker run -d --network host quay.io/coreos/etcd:v3.3
```
[the remark this answers is not in the document]
Comment (binoue) [reply .github/workflows/main.yaml:23]: Inside of the container `quay.io/coreos/etcd:v3.3`, actions/setup-go@v2 did not work well.
It is still possible to install go in the container manually, but I think that is not worth doing.

File: .github/workflows/main.yaml [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        run: docker build -t quay.io/cybozu/cke:latest ./docker
+  mtest:
+    name: Mtest
+    runs-on: ubuntu-20.04
+    needs: build
+    strategy:
+      matrix:
+        suite: [functions, robustness, operators]
+    env:
+      INSTANCE_NAME: cke-${{ matrix.suite }}-${{ github.run_number }}
```
[the remark this answers is not in the document]
Comment (binoue) [reply .github/workflows/main.yaml]: Thank you. I'll fix it as you suggested.

File: .github/workflows/main.yaml (hunk 1 above)
Comment (binoue) [reply .github/workflows/main.yaml]: Thank you. I'll fix it as you suggested.

File: .github/workflows/main.yaml (hunk 2 above)
Comment (binoue) [reply .github/workflows/main.yaml:53]: Thank you. I'll fix this

File: Makefile [hunk 9]
```diff
@@ -38,3 +38,15 @@ nilerr:
 	if ! which nilerr >/dev/null; then \
 		env GOFLAGS= go install github.com/gostaticanalysis/nilerr/cmd/nilerr@latest; \
 	fi
+
+.PHONY: goimports
+goimports:
+	if ! which goimports >/dev/null; then \
+		env GOFLAGS= go install golang.org/x/tools/cmd/goimports; \
```
Comment (ymmt2005) [inline Makefile:45]:
```suggestion
		env GOFLAGS= go install golang.org/x/tools/cmd/goimports@latest; \
```

File: Makefile [hunk 10]
```diff
@@ -38,3 +38,15 @@ nilerr:
...
+	fi
+
+.PHONY: custom-checker
+custom-checker:
+	if ! which custom-checker >/dev/null; then \
+		env GOFLAGS= go install github.com/cybozu/neco-containers/golang/analyzer/cmd/custom-checker; \
```
Comment (ymmt2005) [inline Makefile:51]: ditto

File: .github/workflows/main.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (binoue) [reply .github/workflows/main.yaml:24]: Sorry for my misunderstanding. Using a service container may work. 
However, in that case, we need to map ports on the host or the service container explicitly.

I don't think its worth doing. (mapping 2379:2379 and 2380:2380 were not enough )

[the remark this answers is not in the document]
Comment (yokaze) [reply .github/workflows/main.yaml:24]: I understand, thank you for the investigation & explanation.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Review by yokaze — APPROVED** [review]
LGTM

**Pull request merged**